x-pack/plugins/security_solution/public/timelines/containers/api.ts

```diff
@@ -242,3 +270,77 @@ export const installPrepackedTimelines = async (): Promise<ImportTimelineResultS
 
   return decodePrepackedTimelineResponse(response);
 };
+
+export const getTimeline = async (id: string) => {
+  const response = await KibanaServices.get().http.get<SingleTimelineResponse>(TIMELINE_URL, {
+    query: {
+      id,
+    },
+  });
+
+  return decodeSingleTimelineResponse(response);
+};
+
+export const getTimelineTemplate = async (templateTimelineId: string) => {
+  const response = await KibanaServices.get().http.get<SingleTimelineResponse>(TIMELINE_URL, {
+    query: {
+      template_timeline_id: templateTimelineId,
+    },
+  });
+
+  return decodeSingleTimelineResponse(response);
+};
+
+export const getAllTimelines = async (args: GetTimelinesArgs, abortSignal: AbortSignal) => {
+  const response = await KibanaServices.get().http.fetch<AllTimelinesResponse>(TIMELINES_URL, {
+    method: 'GET',
+    query: {
+      ...(args.onlyUserFavorite ? { only_user_favorite: args.onlyUserFavorite } : {}),
+      ...(args?.pageInfo?.pageSize ? { page_size: args.pageInfo.pageSize } : {}),
+      ...(args?.pageInfo?.pageIndex ? { page_index: args.pageInfo.pageIndex } : {}),
+      ...(args.search ? { search: args.search } : {}),
+      ...(args?.sort?.sortField ? { sort_field: args?.sort?.sortField } : {}),
+      ...(args?.sort?.sortOrder ? { sort_order: args?.sort?.sortOrder } : {}),
+      ...(args.status ? { status: args.status } : {}),
+      ...(args.timelineType ? { timeline_type: args.timelineType } : {}),
+    },
+    signal: abortSignal,
+  });
+
+  return decodeAllTimelinesResponse(response);
+};
+
+export const persistFavorite = async ({
+  timelineId,
+  templateTimelineId,
+  templateTimelineVersion,
+  timelineType,
+}: {
+  timelineId?: string | null;
+  templateTimelineId?: string | null;
+  templateTimelineVersion?: number | null;
+  timelineType: TimelineType;
+}) => {
+  const response = await KibanaServices.get().http.patch<ResponseFavoriteTimeline>(
+    TIMELINE_FAVORITE_URL,
+    {
+      method: 'PATCH',
+      body: JSON.stringify({
```

Comment:
nit -> to be safe we should wrap the `json.stringify` around a `try/catch` and `throw` an error if something is bad